src/firmware/app/control/trajectory_planner.h

```diff
@@ -261,11 +279,21 @@ void static app_trajectory_planner_generateTimeProfile(
  *
  *  TODO: Remove when #1322 is merged.
  *
- * @pre forwards is pre-allocated up to at least TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
- * @param forwards This is the trajectory that will be modified in place to become a
- * reverse trajectory
- * @param num_segments The number of segements(elements) in the forwards array
+ * @param forwards_trajectory This is the trajectory that will be modified in place to
+ * become a reverse trajectory
  */
 void static app_trajectory_planner_reverseTrajectoryDirection(
-    TrajectoryElement_t forwards[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const unsigned int num_segments);
+    PositionTrajectory_t* forwards_trajectory);
+
+/***
```

Comment:
Nothing sneaks past MatLint:tm:  :eyes: 
```suggestion
/**
```